.github/workflows/containerized-ci.yml

```diff
@@ -1,84 +1,123 @@
-    - name: Environmental variables - nvfortran
-      if: contains(matrix.container, 'nvfortran')
-      run: echo "FCFLAGS=-Mallocatable=03 -Mstandard -Mbounds -Mchkptr -Kieee -Mchkstk" >> $GITHUB_ENV
-
-    - name: Make library, examples, tests
-      shell: bash
-      env:
-        RTE_KERNELS: ${{ matrix.rte-kernels }}
+    #
+    # Checks-out repository under $GITHUB_WORKSPACE
+    #
+    - uses: actions/checkout@v3
+    #
+    # Install Zstd
+    #
+    - name: Install Zstd
       run: |
-        source /opt/intel/oneapi/setvars.sh || true
-        cd ${RRTMGP_ROOT}
-        ${FC} --version
-        make libs
-        make -C build separate-libs
-    ############################################################################
+        # We skip 'apt-get update' to keep this step lightweight. However, that
+        # might lead to a failure, which we tolerate because the only reason we
+        # run it is to make cache files compatible across workflows.
+        # TODO: update the containers so that we could skip the following:
+        apt-get install zstd || true
+    #
+    # Cache RFMIP files
+    #
     - name: Cache RFMIP files
-      id: cache-rfmip-files
       uses: actions/cache@v3
       with:
-        path: /home/runner/rfmip-files # Same as #{RFMIP_DIR}
+        path: ${{ env.RFMIP_CACHEDIR }}
         key: rfmip-files
-
+    #
+    # Stage RFMIP files
+    #
     - name: Stage RFMIP files
-      if: steps.cache-rfmip-files.outputs.cache-hit != 'true'
       run: |
-        mkdir -p ${RFMIP_DIR}
-        cd ${RFMIP_DIR}
-        python ${RRTMGP_ROOT}/examples/rfmip-clear-sky/stage_files.py
-    ############################################################################
-    - name: Run examples, tests
-      shell: bash
-      env:
-        LD_LIBRARY_PATH: /home/runner/netcdf-c/lib
+        if test ! -d "${RFMIP_CACHEDIR}"; then
+          mkdir -p "${RFMIP_CACHEDIR}" && cd "${RFMIP_CACHEDIR}"
+          python "${GITHUB_WORKSPACE}/examples/rfmip-clear-sky/stage_files.py"
+        fi
+        for file in "${RFMIP_CACHEDIR}"/*; do
+          if test ! -d "${file}"; then
+            echo "copying '${file}'..."
+            cp "${file}" "${GITHUB_WORKSPACE}/examples/rfmip-clear-sky/"
+          fi
+        done
+    #
+    # Finalize build environment
+    #
+    - name: Finalize build environment
+      if: contains('ifort ifx', matrix.fortran-compiler)
       run: |
-        source /opt/intel/oneapi/setvars.sh || true
-        export LD_LIBRARY_PATH=${NFHOME}/lib:${LD_LIBRARY_PATH}
-        make tests
-    - name: Comparison
-      run: make check
-    ############################################################################
-    - name: Validation plots
-      if: contains(matrix.container, 'ifort') && contains(matrix.rte-kernels, 'default')
+        # TODO: update the Intel container so that we do not have to the following:
+        echo "BASH_ENV=/opt/intel/oneapi/setvars.sh" >> "${GITHUB_ENV}"
```

Comment:
@skosukhin Can you please explain precisely what should be done in the Intel container to be able to remove this step?

Reply:
It could be something like `RUN echo ". /opt/intel/oneapi/setvars.sh" >> /etc/profile.d/oneapi.sh` in the Dockerfile but this is not needed after switching to the official Intel image.